great_expectations/expectations/expectation.py

```diff
@@ -1632,20 +1634,46 @@ def _validate_metric_value_between(  # noqa: C901, PLR0912
         return {"success": success, "result": {"observed_value": metric_value}}
 
 
+UNEXPECTED_ROWS_EXPECTATION_SHORT_DESCRIPTION = (
+    "This Expectation will fail validation if the query returns one or more rows.  "
+    "The WHERE clause defines the fail criteria."
+)
+UNEXPECTED_ROWS_QUERY_DESCRIPTION = "A SQL or Spark-SQL query to be executed for validation."
+SUPPORTED_DATA_SOURCES = ["SQL", "Spark-SQL"]
+
+
 class UnexpectedRowsExpectation(BatchExpectation):
-    """
-    UnexpectedRowsExpectations facilitate the execution of SQL or Spark-SQL queries as the core logic for an Expectation.
+    __doc__ = f"""{UNEXPECTED_ROWS_EXPECTATION_SHORT_DESCRIPTION }
+
+    UnexpectedRowsExpectations facilitate the execution of SQL or Spark-SQL queries \
+    as the core logic for an Expectation. UnexpectedRowsExpectations must implement \
+    a `_validate(...)` method containing logic for determining whether data returned \
+    by the executed query is successfully validated. One is written by default, but \
+    can be overridden.
 
-    UnexpectedRowsExpectations must implement a `_validate(...)` method containing logic for determining whether data returned by the executed query is successfully validated.
-    One is written by default, but can be overridden.
     A successful validation is one where the unexpected_rows_query returns no rows.
 
+    UnexpectedRowsExpectation is a \
+    [Batch Expectation](https://docs.greatexpectations.io/docs/guides/expectations/creating_custom_expectations/how_to_create_custom_batch_expectations).
+
+    BatchExpectations are one of the most common types of Expectation.
+    They are evaluated for an entire Batch, and answer a semantic question about the Batch itself.
+
     Args:
-        unexpected_rows_query (str): A SQL or Spark-SQL query to be executed for validation.
-    """  # noqa: E501
+        unexpected_rows_query (str): {UNEXPECTED_ROWS_QUERY_DESCRIPTION}
+
+    Returns:
+        An [ExpectationSuiteValidationResult](https://docs.greatexpectations.io/docs/terms/validation_result)
+
+    Supported Datasources:
+        [{SUPPORTED_DATA_SOURCES[0]}](https://docs.greatexpectations.io/docs/application_integration_support/)
+        [{SUPPORTED_DATA_SOURCES[1]}](https://docs.greatexpectations.io/docs/application_integration_support/)
+    """
```

Comment:
Nice work on the docstring! I would remove the supported datasource blob for now. Docs team can add this if they want it.